src/rendergraph/scenegraph/backend/material.cpp

```diff
@@ -0,0 +1,28 @@
+#include "rendergraph/material.h"
+
+#include "rendergraph/materialshader.h"
+
+using namespace rendergraph::backend;
+
+bool Material::updateUniformsByteArray(QByteArray* buf) {
+    auto pThis = static_cast<rendergraph::Material*>(this);
```

Comment:
No, it is of type rendergraph::backend::Material, we need to cast it to its derived class rendergraph::Material here. (A static cast is justified, because the backend class should only be instantiated as base class for the frontend class)

Now, this confusion raises the question: is using a namespace for the backend base classes a good idea? It might be more readable to use different class names, e.g. MaterialBackend or BackendMaterial, or BaseMaterial. What do you think?

Reply:
maybe just disambiguate it from the backend by adding `rendergraph::common`?

Reply:
I think it is a good idea to use a different name for the base class like base "Base" or any more descriptive name.

Reply:
There are already classes with that in the name though...

Reply:
In https://github.com/mixxxdj/mixxx/pull/13599/commits/0a0ebdc36307a8dd6c4eaf3aa2e7805968fb9815 I renamed BaseNode to TreeNode and replace the backend namespace with a Base.. prefix. It definitely makes things more clear.

Please resolve if you are okay with the change.